From 9cee2ce25a98adeedd230983dba2c9e12e456d01 Mon Sep 17 00:00:00 2001
From: Aloka Dixit <alokad@codeaurora.org>
Date: Wed, 16 Sep 2020 15:54:56 -0700
Subject: [PATCH] auth: Add support for Opportunistic Wireless Encryption

This patch adds support for OWE in wireless configuration file.
Use 'option encryption owe'.

Signed-off-by: Aloka Dixit <alokad@codeaurora.org>

--- a/scripts/netifd-wireless.sh	2023-04-14 19:55:14.387934326 -0700
+++ b/scripts/netifd-wireless.sh	2023-04-14 19:57:30.142773168 -0700
@@ -85,6 +85,9 @@
 		*sae*)
 			auth_type=sae
 		;;
+		owe)
+			auth_type=owe
+		;;
 	esac
 }
 
@@ -233,7 +236,7 @@
 	case "$encryption" in
 		*tkip+aes|*tkip+ccmp|*aes+tkip|*ccmp+tkip) wpa_cipher="CCMP TKIP";;
 		*ccmp256) wpa_cipher="CCMP-256";;
-		*aes|*ccmp| psk2 | wpa2 | sae*) wpa_cipher="CCMP";;
+		*aes|*ccmp| psk2 | wpa2 | sae* | owe) wpa_cipher="CCMP";;
 		*tkip | wpa | psk ) wpa_cipher="TKIP";;
 		*gcmp256) wpa_cipher="GCMP-256";;
 		*gcmp) wpa_cipher="GCMP";;
